spec/services/set_decline_by_default_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+
+RSpec.describe SetDeclineByDefault do
+  describe '#call' do
+    let(:application_form) { create(:completed_application_form) }
+    let(:choices) { application_form.application_choices }
+    let(:result) { SetDeclineByDefault.new(application_form: application_form).call }
+    let(:time_limit_calculator) { double(:TimeLimitCalculator, call: 10) }
+
+    before { allow(TimeLimitCalculator).to receive(:new).and_return(time_limit_calculator) }
+
+    def expect_all_decline_by_default_at_values_to_be(expected_value)
+      application_form.application_choices.each do |application_choice|
+        expect(application_choice.decline_by_default_at).to eq(expected_value)
+      end
+    end
+
+    context 'all the application choices have an offer' do
+      it 'the DBD is set to 10 business days from the date of the most recent offer' do
+        choices[0].update(status: :offer, offered_at: 1.business_days.from_now)
+        choices[1].update(status: :offer, offered_at: 2.business_days.from_now)
+        choices[2].update(status: :offer, offered_at: 3.business_days.from_now)
```

Comment:
seems a it odd to have the offer dates in the future, though it's logically correct I guess

Reply:
I agree, has been changed